pkg/network/ovn_kubernetes.go

```diff
@@ -532,10 +548,23 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootstrap.BootstrapResult) (renderIPsecMachineConfig, renderIPsecDaemonSet,
 	renderIPsecOVN, renderIPsecHostDaemonSet, renderIPsecContainerizedDaemonSet, renderIPsecDaemonSetAsCreateWaitOnly bool) {
 	isHypershiftHostedCluster := bootstrapResult.Infra.HostedControlPlane != nil
-	isIPsecEnabled := conf.IPsecConfig != nil
 	isIpsecUpgrade := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.LegacyIPsecUpgrade
 	isOVNIPsecActive := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.OVNIPsecActive
 
+	// Find the IPsec mode from Ipsec.config
+	// Ipsec.config == nil (bw compatibility) || ipsecConfig == Off ==> ipsec is disabled
+	// ipsecConfig.mode == "" (bw compatibility) || ipsec.Config == Full ==> ipsec is enabled for NS and EW
+	// ipsecConfig.mode == External ==> ipsec is enabled for NS only
+
+	mode := getIPsecMode(conf)
+	isIPsecEnabled := mode != operv1.IPsecModeDisabled
+	isIPsecFull := mode == operv1.IPsecModeFull // Full mode is for NS+EW
+	if conf.IPsecConfig != nil {
+		klog.Infof("IPsec: after looking at %+v, ipsec mode=%s, isIPsecEnabled=%v", conf.IPsecConfig, mode, isIPsecEnabled)
+	} else {
+		klog.V(5).Infof("IPsec: after looking at %+v, ipsec mode=%s, isIPsecEnabled=%v", conf.IPsecConfig, mode, isIPsecEnabled)
+	}
```

Comment:
I guess that this has been useful for debugging but I would refine this comment a bit
```
if conf.IPsecConfig == nil || conf.IPsecConfig.Mode == "" { 
    klog.V(5).Infof("Assuming IPsec mode %s after IPsec config %+v", mode, conf.IPsecConfig)
}
```
It wouldn't harm if this lived in `getIPsecMode` at that log level.